server/server.go

```diff
@@ -462,139 +419,6 @@ func (r *rodeServer) CreateNote(ctx context.Context, request *pb.CreateNoteReque
 	})
 }
 
-type genericListOptions struct {
-	index         string
-	filter        string
-	query         *esutil.EsSearch
-	pageSize      int32
-	pageToken     string
-	sortDirection esutil.EsSortOrder
-	sortField     string
-}
-
-func (r *rodeServer) genericList(ctx context.Context, log *zap.Logger, options *genericListOptions) (*esutil.EsSearchResponseHits, string, error) {
-	body := &esutil.EsSearch{}
-	if options.query != nil {
-		body = options.query
-	}
-
-	if options.filter != "" {
-		log = log.With(zap.String("filter", options.filter))
-		filterQuery, err := r.filterer.ParseExpression(options.filter)
-		if err != nil {
-			return nil, "", createError(log, "error while parsing filter expression", err)
-		}
-
-		body.Query = filterQuery
-	}
-
-	if options.sortField != "" {
-		body.Sort = map[string]esutil.EsSortOrder{
-			options.sortField: options.sortDirection,
-		}
-	}
-
-	searchOptions := []func(*esapi.SearchRequest){
-		r.esClient.Search.WithContext(ctx),
-	}
-
-	var nextPageToken string
-	if options.pageToken != "" || options.pageSize != 0 { // handle pagination
-		next, extraSearchOptions, err := r.handlePagination(ctx, log, body, options.index, options.pageToken, options.pageSize)
-		if err != nil {
-			return nil, "", createError(log, "error while handling pagination", err)
-		}
-
-		nextPageToken = next
-		searchOptions = append(searchOptions, extraSearchOptions...)
-	} else {
-		searchOptions = append(searchOptions,
-			r.esClient.Search.WithIndex(options.index),
-			r.esClient.Search.WithSize(maxPageSize),
-		)
-	}
-
-	encodedBody, requestJson := esutil.EncodeRequest(body)
-	log = log.With(zap.String("request", requestJson))
-	log.Debug("performing search")
-
-	res, err := r.esClient.Search(
```

Comment:
I think this was the last use of `esClient` in the server directly if you want to remove that from the Rode server struct as well

Reply:
there are a few things I missed actually.  it was used in the tests and with the fake transport, which also isn't needed anymore, but there were some assertions left behind so I kept it in.  But I'm working on removing those now

Reply:
I think I got everything if you want to take another look